src/lib.rs

```diff
@@ -36,5 +38,361 @@ pub fn y_py(_py: Python, m: &PyModule) -> PyResult<()> {
+                            _ => {
+                                panic!("Unexpected type ref: {:?}", ptr.type_ref());
+                            }
+                        }
+                    }
+                    ItemContent::String(child_text_part) => {
+                        if !child_result.is_empty() {
+                            let mut child_text = child_result
+                                .get("text")
+                                .unwrap_or(&Any::String("".to_string().into()))
+                                .to_string();
+                            child_text.push_str(child_text_part.as_str());
+                            child_result.insert("text".to_string(), Any::String(child_text.into()));
+                        }
+                    }
+                    _ => {
+                        eprintln!("Ignored child of XmlTextRef: {:?}", child);
+                    }
+                }
+            }
+            if !child_result.is_empty() {
+                children.push(Any::Map(Box::new(child_result)));
+            }
+            if children.len() > 0 {
+                result.insert(
+                    "children".to_string(),
+                    Any::Array(children.into_boxed_slice()),
+                );
+            }
+        }
+        Any::Map(Box::new(result))
+    }
+
+    fn process_xml_node(
+        txn: &Transaction,
+        result: &mut HashMap<String, Any>,
+        first_child_maybe: Option<XmlNode>,
+    ) -> () {
+        let first_child = match first_child_maybe {
+            Some(first_child) => first_child,
+            None => {
+                return;
+            }
+        };
+        let mut children: Vec<Any> = vec![];
+        match first_child {
+            XmlNode::Text(text) => {
+                children.push(process_xml_text_node(txn, &text));
+                text.siblings(txn)
+                    .for_each(|sibling: XmlNode| match sibling {
+                        XmlNode::Text(text) => {
+                            children.push(process_xml_text_node(txn, &text));
+                        }
+                        XmlNode::Fragment(fragment) => {
+                            eprintln!("Unhandled Fragment: {:?}", fragment);
+                        }
+                        XmlNode::Element(element) => {
+                            eprintln!("Unhandled Element: {:?}", element);
+                        }
+                    });
+            }
+            XmlNode::Fragment(fragment) => {
+                eprintln!("Unhandled Fragment: {:?}", fragment);
+            }
+            XmlNode::Element(element) => {
+                eprintln!("Unhandled Element: {:?}", element);
+            }
+        }
+        result.insert(
+            "children".to_string(),
+            Any::Array(children.into_boxed_slice()),
+        );
+    }
+
+    #[test]
+    fn parse_nested_xml_text_nodes() {
+        const RAW_LEXICAL_STATE_AS_UPDATE: &[u8] = &[
```

Comment:
Storing this as a string is a bit cleaner (one line instead of tons) imo but not a big deal

Reply:
Left if as `Vec<u8>` for now